diff --git a/src/store/db/schema.rs b/src/store/db/schema.rs
index 79df3db..6cb3c71 100644
--- a/src/store/db/schema.rs
+++ b/src/store/db/schema.rs
@@ -217,9 +217,27 @@ diesel::table! {
     }
 }

-diesel::joinable!(catalog -> storage (storage));
-diesel::joinable!(media_person -> person (catalog));
-diesel::joinable!(media_search -> saved_search (catalog));
+diesel::table! {
+    user_catalog (user, catalog) {
+        user -> Varchar,
+        catalog -> Varchar,
+        writable -> Bool,
+    }
+}
+
+diesel::table! {
+    album_descendent (id, descendent) {
+        id -> Varchar,
+        descendent -> Varchar,
+    }
+}
+
+diesel::table! {
+    tag_descendent (id, descendent) {
+        id -> Varchar,
+        descendent -> Varchar,
+    }
+}

 diesel::allow_tables_to_appear_in_same_query!(
     album,
@@ -234,6 +252,9 @@ diesel::allow_tables_to_appear_in_same_query!(
     person,
     saved_search,
     shared_catalog,
+    user_catalog,
+    album_descendent,
+    tag_descendent,
     storage,
     tag,
     user,
