diff --git b/packages/service/src/store/db/schema.rs a/packages/service/src/store/db/schema.rs
index 886ae32..0c2dd54 100644
--- b/packages/service/src/store/db/schema.rs
+++ a/packages/service/src/store/db/schema.rs
@@ -1,5 +1,7 @@
 // @generated automatically by Diesel CLI.
 
+pub use super::views::*;
+
 pub mod sql_types {
     #[derive(diesel::sql_types::SqlType)]
     #[diesel(postgres_type(name = "location"))]
@@ -231,6 +233,15 @@ diesel::joinable!(media_person -> person (catalog));
 diesel::joinable!(media_search -> saved_search (catalog));
 
 diesel::allow_tables_to_appear_in_same_query!(
+    album_descendent,
+    album_relation,
+    latest_media_file,
+    media_file_alternates,
+    person_relation,
+    tag_descendent,
+    tag_relation,
+    search_relation,
+    user_catalog,
     album,
     alternate_file,
     auth_token,
